modules/seqexec/server/src/main/scala/seqexec/server/SeqexecEngine.scala

```diff
@@ -105,22 +107,24 @@ class SeqexecEngine(httpClient: Client[IO], gpi: GpiClient[IO], ghost: GhostClie
 
   def sync(q: EventQueue, seqId: Observation.Id): IO[Either[SeqexecFailure, Unit]] =
     odbLoader.loadEvents(seqId).flatMap{ e =>
-      q.enqueue(Stream.emits(e)).map(_.asRight).compile.last.attempt.map(_.bimap(SeqexecFailure.SeqexecException.apply, _ => ()))
+      q.enqueue(
+        Stream.emits(e)).map(_.asRight).compile.last.attempt.map(_.bimap(SeqexecFailure.SeqexecException.apply, _ => ())
+      )
     }
 
-  // TODO: this is too much guessing. We should have proper tracking of systems' state.
-  def failedInstruments(st: EngineState): Set[Resource] = st.sequences.values.toList.mapFilter(s =>
-    s.seq.status.isError.option(s.seqGen.instrument)).toSet
-
   private def checkResources(seqId: Observation.Id)(st: EngineState): Boolean = {
     // Resources used by running sequences
     val used = resourcesInUse(st)
 
     // Resources that will be used by sequences in running queues
-    val reservedByQueues = resourcesReserved(failedInstruments(st), st)
-
-    st.sequences.get(seqId).exists(_.seqGen.resources.intersect(used ++ reservedByQueues).isEmpty)
+    val reservedByQueues = resourcesReserved(st)
 
+    st.sequences.get(seqId).exists(x =>
```

Comment:
I guess this is the core change right?. looks good

Reply:
Yes, I check if the sequence is in the queue, in which case the next one, the check against the resources reserved by the running queue, does not apply.